TWTValidation/JSON Validator/Parser/TWTJSONSchemaParser.m

```diff
@@ -100,13 +104,16 @@ - (TWTJSONSchemaTopLevelASTNode *)parseWithError:(NSError *__autoreleasing *)out
     NSArray *referenceNodes = [topLevelNode allReferenceNodes];
     if (referenceNodes.count > 0) {
         for (TWTJSONSchemaReferenceASTNode *referenceNode in referenceNodes) {
-            TWTJSONSchemaASTNode *referent = [topLevelNode nodeForReferenceNode:referenceNode];
+            TWTJSONSchemaASTNode *referent = (referenceNode.filePath ?
+                                              [self.remoteSchemaManager remoteNodeForReferenceNode:referenceNode] :
+                                              [topLevelNode nodeForReferenceNode:referenceNode]);
 
             if (referent) {
                 referenceNode.referentNode = referent;
             } else {
                 if (outError) {
-                    NSString *description = [NSString stringWithFormat:@"Reference path %@ is invalid and/or does not match this schema.", [referenceNode.referencePathComponents componentsJoinedByString:@"/"]];
+                    NSString *fullReferencePath = [[@"" stringByAppendingString:referenceNode.filePath] stringByAppendingString:[referenceNode.referencePathComponents componentsJoinedByString:@"/"]];
```

Comment:
`@""` is to avoid starting with nil. I moved the method to the reference node and tried to structure it so that's more clear.